### PR TITLE
{CI} Notify batch CI errors to teams channel

### PR DESCRIPTION
**Pull request by wangzelin007**
**Related command**

**Description**

 In batch ci mode, if any job failure is detected, sent the message to the teams channel.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
CI

File: scripts/ci/notify_ci_errors.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        response = requests.request("POST", teams_api_url, headers=headers, data=data)
+        logger.debug('Status Code: %s', response.status_code)
+        logger.debug('Response Content: %s', response.content)
+    else:
+        logger.error('Missing variables: \nBASE_URI: %s, PROJECT_TYPE: %s, '
+                     'BUILD_ID: %s, JOB_ID: %s', base_uri, project_type, build_id, job_id)
+
+
+if __name__ == '__main__':
+    notify_batch_ci_errors()
```
Comment (bebound) [inline scripts/ci/notify_ci_errors.py:58]: How about add a newline at the end of file?

**Review by bebound — DISMISSED** [review]

File: azure-pipelines-full-tests.yml [hunk 2]
```diff
@@ -140,3 +140,37 @@ jobs:
...
+      versionSpec: 3.10
+  - task: AzureCLI@2
+    inputs:
+      azureSubscription: 'Azure CLI'
+      scriptType: 'bash'
+      scriptLocation: 'inlineScript'
+      inlineScript: |
+        pip install requests
+        teams_api_url=$(az keyvault secret show --vault-name kv-azuresdk --name azure-cli-teams-bot-api-url --query value -otsv)
+        teams_api_key=$(az keyvault secret show --vault-name kv-azuresdk --name azure-cli-teams-bot-api-key --query value -otsv)
```
Comment (kairu-ms) [inline azure-pipelines-full-tests.yml:168]: I think we should get the vault-name and secret-name from the environment. For security concern.

File: scripts/ci/notify_ci_errors.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    if all([base_uri, project_type, build_id, job_id]):
+        # https://dev.azure.com/azclitools/public/_build/results?buildId=45514&view=logs&j=15eab87b-4a33-5480-11eb-66f5d5b3681b
+        url = f'{base_uri}{project_type}/_build/results?buildId={build_id}&view=logs&j={job_id}'
+
+        data = {
+            "title": "Batch CI Error Appears!",
+            "body": "Azure cli team,\n\nPlease click to take a look at the batch CI error.",
+            "notificationUrl": url,
+            "targetType": "channel",
+            "recipients": "19%3a42c79e58f56149d8917489b0c595fc78%40thread.tacv2"
```
Comment (jiasli) [inline scripts/ci/notify_ci_errors.py:43]: I am not sure if this should be exposed to the public.

**Review by kairu-ms — DISMISSED** [review]

File: azure-pipelines-full-tests.yml (hunk 2 above)
Comment (wangzelin007) [reply azure-pipelines-full-tests.yml:168]: ADO will prevent pull request from fork repo trigger jobs that contain sensitive information.
https://learn.microsoft.com/en-us/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#contributions-from-forks
image

File: scripts/ci/notify_ci_errors.py (hunk 3 above)
Comment (wangzelin007) [reply scripts/ci/notify_ci_errors.py]: Save teams_channel_id to secret variable.

**Pull request merged**